pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/generators/ModelGenerator.java

```diff
@@ -253,77 +131,155 @@ public void generate(final Protobuf3Parser.MessageDefContext msgDef,
+
+		// oneof enums
+		bodyContent += String.join("\n    ", oneofEnums);
+
+		// === Build file
+		try (final FileWriter javaWriter = new FileWriter(javaFile)) {
+			javaWriter.write(
+					generateClass(modelPackage, imports, javaDocComment, deprecated, javaRecordName, fields, bodyContent)
+			);
+		}
+	}
+
+	/**
+	 * Generating method that assembles all the previously generated pieces together
+	 * @param modelPackage the model package to use for the code generation
+	 * @param imports the imports to use for the code generation
+	 * @param javaDocComment the java doc comment to use for the code generation
+	 * @param deprecated the deprecated annotation to add
+	 * @param javaRecordName the name of the class
+	 * @param fields the fields to use for the code generation
+	 * @param bodyContent the body content to use for the code generation
+	 * @return the generated code
+	 */
+	@NotNull
+	private static String generateClass(final String modelPackage,
+										final Set<String> imports,
+										final String javaDocComment,
+										final String deprecated,
+										final String javaRecordName,
+										final List<Field> fields,
+										final String bodyContent) {
+		return """
+				package $package;
+				$imports
+				import com.hedera.pbj.runtime.Codec;
+				import java.util.function.Consumer;
+				import java.lang.Comparable;
+				import edu.umd.cs.findbugs.annotations.Nullable;
+				import static java.util.Objects.requireNonNull;
+									
+				$javaDocComment$deprecated
+				public record $javaRecordName(
+				$fields) implements Comparable<$javaRecordName> {
+				$bodyContent}
+				"""
+				.replace("$package", modelPackage)
+				.replace("$imports", imports.isEmpty() ? "" : imports.stream().collect(Collectors.joining(".*;\nimport ", "\nimport ", ".*;\n")))
+				.replace("$javaDocComment", javaDocComment)
+				.replace("$deprecated", deprecated)
+				.replace("$javaRecordName", javaRecordName)
+				.replace("$fields", fields.stream().map(field ->
+						(field.type() == FieldType.MESSAGE ? "@Nullable " : "")
+								+ field.javaFieldType() + " " + field.nameCamelFirstLower()
+				).collect(Collectors.joining(",\n")).indent(DEFAULT_INDENT))
+				.replace("$bodyContent", bodyContent);
+	}
+
+
+	/**
+	 * Filter the fields to only include those that are comparable
+	 * @param msgDef The message definition
+	 * @param lookupHelper The lookup helper
+	 * @param fields The fields to filter
+	 * @return the filtered fields
+	 */
+	@NonNull
+	private static List<Field> filterComparableFields(final MessageDefContext msgDef,
+													  final ContextualLookupHelper lookupHelper,
+													  final List<Field> fields) {
+		final Set<String> comparableFields = lookupHelper.getComparableFields(msgDef);
+		if(comparableFields.isEmpty()) {
+			// consider all fields in compareTo by defalut
+			return fields;
+		}
+		return fields.stream().filter(f -> comparableFields.contains(f.nameCamelFirstLower())).collect(Collectors.toList());
```

Comment:
Now that I got here, I see why you used Sets above. However, for the reasons from previous comments, I'd suggest to use Lists there, and then this method wouldn't be needed at all. This is because the model owner may want to specify a particular order of fields for the purpose of comparison, which may not always be identical to the order in which the fields are defined in the model (based on the business use-case etc.)